### DISCUSSION TITLE
CUPS for printer server, Samsung SCX-3405W

### DISCUSSION THREAD

**Discussion by KaptenLada** [OpenPrinting/cups#840] [Q&A]
So I was trying to fix a printer than an old couple had problems with and they were using a chrome os computer and it didn't have the SCX-3405W driver that I needed to make it work. I tried making it a generic printer and also tried to wirelessly connect it without the usb cable. didn't work either. So I'm wondering if making a printer server would work for the printer with a RPI zero? I saw that CUPS didn't have any drivers for SCX-3405w. But it shows up [here](https://openprinting.github.io/printers/) so maybe it works. So yeah I'm just wondering if a CUPS print server would work or if I just need to buy a new printer.

The printer worked with windows and usb cable btw.

**Comment by dkosovic** [comment]
HP acquired the Samsung Printer Business a number of years ago.

AirPrint and Mopria are extensions of IPP Everywhere, which are driverless printing implementations.

The following HP page shows how to enable AirPrint and Bonjour(mDNS) on Samsung printers:
https://support.hp.com/us-en/document/c05587132

CUPS and ChromeOS support IPP Everywhere.

Once you enable AirPrint and Bonjour(mDNS) on the printer, CUPS and ChromeOS should be able to find it and use IPP Everywhere instead of a driver.

**Reply by michaelrsweet** [reply]
@dkosovic AirPrint is a separate "profile" of IPP that came before and is very similar to IPP Everywhere, while Mopria came after, starting with IPP Everywhere and adding a few things. The IPP Everywhere "driver" in CUPS and ChromeOS works with all of the "driverless" IPP standards/profiles.

**Reply by KaptenLada** [reply]
So i dont need an rpi zero for it to work?

**Reply by dkosovic** [reply]
If the Samsung SCX-3405w supports PDF printing or one of the raster formats ChromeOS supports, you shouldn't need a RPI Zero to do IPP printing from ChromeOS to that AirPrint printer. There is a chance that the SCX-3405w only supports Apple Raster (URF) and not PDF nor one of the other raster formats. I think ChromeOS doesn't support URF. Sorry I may have misled you in not requiring a RPI Zero.

Make sure you use the latest version of Raspberry Pi OS (which was released in October of this year and based on Debian 12 Bookworm) as the version of CUPS that it comes with has proper support for AirPrint. The CUPS server will be able to convert PDF received from ChromeOS to the URF output sent to the printer.

**Reply by dkosovic** [reply]
Forgot to mention, the following Debian CUPS driverless printing page might be a good reference page: 
* https://wiki.debian.org/CUPSDriverlessPrinting

You can use the `ippfind` command to find the printer and determine its IPP URL, then issue something like the following to add the discovered IPP URL as an IPP Everywhere print queue:
```
lpadmin -p SCX-3405w -v ipp://xxx.xxx.xxx.xxx/ipp/print -E -m everywhere
```

**Reply by KaptenLada** [reply]
Wait so just to be sure, i need to have the rpi zero to make it work ig?

**Reply by dkosovic** [reply]
I'm really not certain about URF support with ChromeOS, I think later versions natively support URF, but there are too many unknowns. They might be using a Chromebook that is no longer supported, so no longer getting ChromeOS updates, etc.

I think the best bet is to try enabling AirPrint and Bonjour on the printer and seeing if you can add the printer in ChromeOS. If ChromeOS can't discover the printer for whatever reason you could try adding an IPP printer manually, one of the parameters is the IP address or hostname of the printer and the other the queue which would most likely be `ipp/print` (some versions of ChromeOS take an IPP URL for the printer that would look like `ipp://xxx.xxx.xxx.xxx/ipp/print`).
